lib/CleantalkSP/SpbctWP/RemoteCalls.php

```diff
@@ -176,31 +176,38 @@ public static function action__download__quarantine_file() // phpcs:ignore PSR1.
      */
     public static function action__update_settings() // phpcs:ignore PSR1.Methods.CamelCapsMethodName.NotCamelCaps
     {
+        if ( ! headers_sent() ) {
+            header('Content-Type: application/json');
+        }
 
         global $spbc;
 
-        foreach ( $spbc->default_settings as $setting ) {
-            if ( Request::get($setting) ) {
-                $var = Request::get($setting);
-                $type = gettype($spbc->settings[$setting]);
+        $incoming_settings = json_decode(Request::get('settings'), true);
 
-                switch ( $type ) {
-                    case 'string':
-                        $var = preg_replace(array('/=/', '/`/'), '', $var);
-                        break;
-                    case 'array':
-                        $var = explode(',', $var);
-                        break;
-                }
+        if ( ! $incoming_settings ) {
+            die(json_encode(['error' => 'No settings provided']));
+        }
+
+        $result = [];
+        foreach ( $spbc->default_settings as $setting_name => $_setting_value ) {
```

Comment:
_setting_value

Reply:
Psalm probably will warn about this unused variable. Can we ban this?

Reply:
Psalm will not warn about variables with underline prefix.